File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    List<Integer> list = new ArrayList<>();
    
    public void traverse(TreeNode root){
    
        if(root==null) return;
        
        traverse(root.left);
        list.add(root.val);
        traverse(root.right);      
        
    }
    
    public boolean isValidBST(TreeNode root) {
        traverse(root);
        
        for(int i=1; i<list.size() ; i++){
            if(list.get(i)<=list.get(i-1)){
                return false;
            }
        }
        
        return true;
    }
}

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        return self.validate(root, float('-inf'), float('inf'))
        
        
    def validate(self, root: Optional[TreeNode], min_val: float, max_val: float)->bool:
        if root==None: return True
        
        if root.val <= min_val or root.val >= max_val:
            return False
        
        return self.validate(root.right, root.val, max_val) and self.validate(root.left, min_val, root.val)
        

        
        
        